service/PayTrace/PayTracePaymentServices.xml

```diff
@@ -0,0 +1,493 @@
+            <implements service="mantle.account.PaymentServices.authorize#Payment"/>
+            <out-parameters>
+                <parameter name="responseMap" type="Map"/>
+            </out-parameters>
+            <actions>
+                <entity-find-one entity-name="mantle.account.payment.Payment" value-field="payment"/>
+                <if condition="payment == null"><return message="Payment ${paymentId} not found"/></if>
+                <set field="paymentMethod" from="payment.'mantle.account.method.PaymentMethod'"/>
+                <if condition="paymentMethod?.paymentMethodTypeEnumId != 'PmtCreditCard'">
+                    <return error="true" message="Cannot authorize and capture payment ${paymentId}, not a Credit Card payment."/></if>
+
+                <if condition="!cardSecurityCode">
+                    <set field="creditCard" from="payment.'mantle.account.method.CreditCard'"/>
+                    <set field="cardSecurityCode" from="creditCard?.cardSecurityCode"/>
+                </if>
+                <if condition="cardSecurityCode == null"><return error="true" message="Cannot authorize payment ${paymentId}, must include a cardSecurityCode."/></if>
+
+                <set field="postalAddress" from="paymentMethod.'mantle.party.contact.PostalAddress'"/>
+               
+                <!-- assemble the full request Map -->
+                <set field="requestMap" from="[amount:payment.amount,credit_card:[number:creditCard?.cardNumber,expiration_month:creditCard?.expireDate ? creditCard.expireDate.substring(0, creditCard.expireDate.indexOf('/')) : null,expiration_year:creditCard?.expireDate ? creditCard.expireDate.substring(creditCard.expireDate.indexOf('/')+1],csc:cardSecurityCode ? cardSecurityCode: '',billing_address:[name:postalAddress?.toName,street_address:postalAddress?.address1,city:postalAddress?.city,state:postalAddress?.stateProvinceGeoId ? postalAddress.stateProvinceGeoId.substring(postalAddress.stateProvinceGeoId.indexOf('_')+1) : null,zip:postalAddress?.postalCode]]"/>
+
+                <entity-find-one entity-name="PayTrace.PaymentGatewayPayTrace" value-field="pgpt"/> 
+                <if condition="pgpt == null"><return error="true" message="No PaymentGatewayPayTrace found with ID ${paymentGatewayConfigId}"/></if>
+
+                <service-call name="PayTrace.PayTracePaymentServices.payTrace#login" in-map="context" out-map="token"/>
+
+                <script><![CDATA[
+                    org.moqui.util.RestClient restClient = ec.service.rest().method(org.moqui.util.RestClient.POST)
+                        .addHeader("Authorization", "bearer ${token.token.access_token}")
+                        .addHeader("Content-Type", "application/json").jsonObject(requestMap)
+                    restClient.uri().protocol("https").host(pgpt.transactionUrl).port(443).path(pgpt.apiVersion).path("transactions").path("sale").path("keyed").build()
```

Comment:
I see no capture happening here. I think this is just auth. The documentation states
"A Capture request can be submitted later to convert this authorization into a Sale, that can be settled for payment." 

I think we need a subsequent call to capture.

Reply:
Hello Daniel, this service is called "authorizeAndCapture" and it helps us to make the authorization and the capture in a single request, according to the api of Paytrace https://developers.paytrace.com/support/home#14000040854